### PR TITLE
Fix #1689 - props mutated in TextInput render method.

### LINKED ISSUE #1689: [TextInput] Not clear what Warning Message for TextInput means.
After upgrading React Native to 0.6.0 I'm starting to get this warning.

`Warning: Don't set .props.style of the React component <TextInput />. Instead, specify the correct value when initially creating the element or use React.cloneElement to make a new element with updated props.`

I did everything as it describes above. but nothing works so far. Any idea why this warning start to show up. I would like to see some example if that possible,

### PR DESCRIPTION
**Pull request by admmasters**
Closes #1689 - mutating props within the render method.

### CODE REVIEW

File: Libraries/Components/TextInput/TextInput.js [hunk 1]
```diff
@@ -411,8 +411,10 @@ var TextInput = React.createClass({
   _renderIOS: function() {
     var textContainer;
 
-    var props = this.props;
-    props.style = [styles.input, this.props.style];
+	  var props = {
+			...this.props,
+		  style: [styles.input, this.props.style]
+    };
```
Comment (ide) [inline Libraries/Components/TextInput/TextInput.js]: can you update this to use spaces instead of tabs?
also put a trailing comma after the last property

Comment (admmasters) [inline Libraries/Components/TextInput/TextInput.js]: No problem - sorry, been away for a few days, will look at doing this shortly.

**Comment by admmasters** [comment]
Fixed a flow warning and updated the PR to no use spaces instead of tabs.

**Comment by admmasters** [comment]
Sorry, my bad, got some issues with Flow - having trouble validating against: 'Text.propTypes.style' on line 286 and not hugely up on spread syntax yet. #66b6e61 breaks Flow.

**Comment by ide** [comment]
@admmasters try `Object.assign({}, this.props)` which I believe is recognized by Flow.

**Comment by admmasters** [comment]
Again - a mad week - I was going to suggest whether or not you wanted me to use Object.assign to close this off :-)

**Pull request closed without merging**